### PR TITLE
[SYCL] Fix ext::oneapi::experimental::joint_reduce 

### PR DESCRIPTION
**Pull request by dm-vodopyanov**
This patch fixes ext::oneapi::experimental::joint_reduce, it missed
the case when WG size is bigger than size of input data.

Test: https://github.com/intel/llvm-test-suite/pull/1452

### CODE REVIEW

**Comment by dm-vodopyanov** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1452

**Comment by dm-vodopyanov** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1452

File: sycl/include/sycl/ext/oneapi/experimental/user_defined_reductions.hpp [hunk 1]
```diff
@@ -41,6 +37,25 @@ reduce_over_group(GroupHelper group_helper, T x, BinaryOperation binary_op) {
...
+reduce_over_group(GroupHelper group_helper, T x, BinaryOperation binary_op) {
+  if constexpr (sycl::detail::is_native_op<T, BinaryOperation>::value) {
+    return sycl::reduce_over_group(group_helper.get_group(), x, binary_op);
+  }
+#ifdef __SYCL_DEVICE_ONLY__
+  return detail::reduce_over_group_impl(
+      group_helper, x, group_helper.get_group().get_local_linear_range(),
+      binary_op);
+#else
+  std::ignore = group_helper;
```
Comment (romanovvlad) [inline sycl/include/sycl/ext/oneapi/experimental/user_defined_reductions.hpp:54]:
```suggestion
```

File: sycl/include/sycl/ext/oneapi/experimental/user_defined_reductions.hpp [hunk 2]
```diff
@@ -41,6 +37,25 @@ reduce_over_group(GroupHelper group_helper, T x, BinaryOperation binary_op) {
                       PI_ERROR_INVALID_DEVICE);
 #endif
 }
+} // namespace detail
```
Comment (romanovvlad) [inline sycl/include/sycl/ext/oneapi/experimental/user_defined_reductions.hpp:40]: Please, add `std::ignore = ` for `x`, `num_elements` and `binary_op` for host compilation.

**Comment by dm-vodopyanov** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1452

**Review by romanovvlad — APPROVED** [review]

**Pull request merged**